submariner/cable-driver-policy.md

```diff
@@ -0,0 +1,327 @@
+  Subnets            []string          `json:"subnets"`
+  PrivateIP          string            `json:"private_ip"`
+  PublicIP           string            `json:"public_ip"`
+  NATEnabled         bool              `json:"nat_enabled"`
+  Backends           []BackendSpec     `json:"backends,omitempty"`
+  SupportedPolicies  []string          `json:"supported_policies,omitempty"`
+}
+
+type BackendSpec struct {
+    CableDriver      CableDriver        `json:"cable_driver,omitempty"`
+}
+
+type CableDriver struct {
+    Name       string                   `json:"name,omitempty"`
+    Config     map[string]string        `json:"config,omitempty"`
+}
+```
+
+The `GatewayStatus` should be updated to reflect the supported backends rather than just one.
+
+```go
+type GatewayStatus struct {
+  Version       string       `json:"version"`
+  HAStatus      HAStatus     `json:"haStatus"`
+  LocalEndpoint EndpointSpec `json:"localEndpoint"`
+  StatusFailure string       `json:"statusFailure"`
+  Connections   []Connection `json:"connections"`
+  Backends      []BackendSpec   `json:"backends"`
+}
+```
+
+The `ConnectionStatus` that reflects the state of different connections on a Submariner gateway
+should be extended to include a new message `conflict` for situations where conflicting policies
+exists for a connection between clusters.
+
+```go
+type ConnectionStatus string
+
+const (
+  Connected          ConnectionStatus = "connected"
+  Connecting         ConnectionStatus = "connecting"
+  ConnectionError    ConnectionStatus = "error"
+  ConnectionConflict ConnectionStatus = "conflict"
+)
+```
+
+The Connection information stored on a Gateway should be updated to reflect what policy is in use for that connection.
+
+```go
+type Connection struct {
+  Status        ConnectionStatus  `json:"status"`
+  StatusMessage string            `json:"statusMessage"`
+  Endpoint      EndpointSpec      `json:"endpoint"`
+  UsingIP       string            `json:"usingIP,omitempty"`
+  UsingNAT      bool              `json:"usingNAT,omitempty"`
+  PolicyID      string            `json:"policy_id,omitempty"`
+  cableType     string            `json:"cable_type,omitempty"`
+  // +optional
+  LatencyRTT *LatencyRTTSpec     `json:"latencyRTT,omitempty"`
+}
+```
+
+### Cluster local Connections Resources
+
+The detailed connection information for a Gateway is currently maintained in the Gateway and Submariner CRs,
+however when the number of clusters grows to a large scale these could be difficult to parse. It maybe preferable
+to maintain connection information in a separate set of `ClusterConnection` CRs. This could be useful in the case of:
+
+- Gateway failover, so the new gateway doesn't need to do the same work again to build a connection definition. Or
+- In the case of `ClusterConnectionPolicy` updates. A Gateway could quickly cycle through the connection policies its
+using in the `ClusterConnection` CRs to see if an update impacts it.
```

Comment:
I'm not clear on the purpose of this. For policy updates, I imagine the GW process would maintain connectivity information in memory in some form to determine the impact of a policy change. In fact I think it could simply re-queue all remote endpoints to re-evaluate them.